pkg/portal/portal.go

```diff
@@ -261,21 +265,41 @@ func (p *portal) unauthenticatedRoutes(r *mux.Router) {
 }
 
 func (p *portal) aadAuthenticatedRoutes(r *mux.Router) {
-	for _, name := range AssetNames() {
-		if name == "v1/build/index.html" {
-			r.NewRoute().Methods(http.MethodGet).Path("/").HandlerFunc(p.index)
-			continue
-		}
+	var names []string
 
-		if name == "v2/build/index.html" {
-			r.NewRoute().Methods(http.MethodGet).Path("/v2").HandlerFunc(p.indexV2)
-			continue
+	err := filepath.Walk(".", func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			log.Fatal(err)
+			return nil
 		}
 
-		fmtName := strings.TrimPrefix(name, "v1/build/")
-		fmtName = strings.TrimPrefix(fmtName, "v2/build/")
+		names = append(names, path)
+		return nil
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, name := range names {
+		regexp, _ := regexp.Compile(`v[1,2]/build/.*\..*`)
+		name := regexp.FindString(name)
+		if len(name) > 0 {
+			if name == "v1/build/index.html" {
+				r.NewRoute().Methods(http.MethodGet).Path("/").HandlerFunc(p.index)
+				continue
+			}
+
+			if name == "v2/build/index.html" {
+				r.NewRoute().Methods(http.MethodGet).Path("/v2").HandlerFunc(p.indexV2)
+				continue
+			}
 
-		r.NewRoute().Methods(http.MethodGet).Path("/" + fmtName).HandlerFunc(p.serve(name))
+			fmtName := strings.TrimPrefix(name, "v1/build/")
+			fmtName = strings.TrimPrefix(fmtName, "v2/build/")
+
+			r.NewRoute().Methods(http.MethodGet).Path("/" + fmtName).HandlerFunc(p.serve(name))
+		}
```

Comment:
```go
     switch name {
     case "v1/build/index.html":
         r.NewRoute().Methods(http.MethodGet).Path("/").HandlerFunc(p.index)
     case "v2/build/index.html":                                                               
         r.NewRoute().Methods(http.MethodGet).Path("/").HandlerFunc(p.indexV2)
     case "":
         //the regex exp did not match and the name is empty
         break                                                           
     default:                                                    
         fmtName := strings.TrimPrefix(name, "v1/build/")
         fmtName = strings.TrimPrefix(fmtName, "v2/build/")
                                                                 
         r.NewRoute().Methods(http.MethodGet).Path("/" + fmtName).HandlerFunc(p.serve(name))
     }                                                           
```

maybe this would be a little better